interface/patient_file/front_payment.php

```diff
@@ -177,6 +177,25 @@ function calcTaxes($row, $amount)
     "i.pid = p.pid AND i.type = 'primary' " .
     "WHERE p.pid = ? ORDER BY i.date DESC LIMIT 1", array($pid));
 
+    /* invoice_no, assuming the encounter items are all billed at once */
+$encrow = sqlQuery("SELECT invoice_refno FROM form_encounter WHERE " .
+    "pid = ? AND encounter = ?", array($pid,$encounter));
+    //$invoice_refno = $encrow['invoice_refno'];
+if(!empty($encrow['invoice_refno'])){
+    $invoice_refno = $encrow['invoice_refno'];
+}
+else{
+    $invoice_refno = sqlQuery("SELECT MAX(invoice_refno) AS invoice_refno FROM form_encounter");
+    if(!empty($invoice_refno['invoice_refno'])){
+        //if invoice_ref is not empty, then increment by 1
+        $invoice_refno= sprintf('%06d', $invoice_refno['invoice_refno'] + 1);
+    }else{
+        //if empty start invoice number at 0000001
+        $invoice_refno= sprintf('%06d', 1);
+
+    }
+}
+
```

Comment:
for above checkout https://github.com/openemr/openemr/blob/master/src/Billing/BillingUtilities.php#L1745-L1749

Reply:
Hello @stephenwaite sorry for the late response. I am not using the easy dev. Is there any alternate approach to getting the styling issues fixed?

> Is this testing well for you? I don't think the demo data has any drugs or treatments added to the billing encounters so can easily test this out.
Yes it is. See the image below.

![Screenshot 2022-08-30  fee sheet demo](https://user-images.githubusercontent.com/112465339/201690684-242e7ad9-f6d1-4625-9452-3ccef26a364d.jpg)

> for above checkout https://github.com/openemr/openemr/blob/master/src/Billing/BillingUtilities.php#L1745-L1749
Also, i do not understand the reference made here. Can you please elaborate?

Reply:
hi @omegasystemsgroup you have to either fix the below manually or could use [phpcbf](https://github.com/squizlabs/PHP_CodeSniffer) with composer on your local to autofix them

you should try to use the existing method in the `BillingUtilities` class so we don't replicate code

![Screenshot from 2022-11-14 11-08-24](https://user-images.githubusercontent.com/5618372/201709831-98814404-9c6f-4431-a917-69ec5dcf0580.png)

Reply:
the `BillingUtilities` method should work fine as long as there's only 1 irnpool defined:

![Screenshot from 2022-11-14 12-25-16](https://user-images.githubusercontent.com/5618372/201725841-7e49cfdd-97c0-4800-b948-a1dde50e4b18.png)